### PR TITLE
Add early stopping support to FloodML

### PR DESCRIPTION
**Pull request by tatianawu**
Added optional early stopping to the FloodML model.

Also fixed a bug in the FloodModelParams dataclass. The default optimizer is being used across all instances of the dataclass that don't specify a different value. Because this is currently a keras.Optimizer object, that same object is getting passed to possibly multiple models, which will throw an error. Changed this to a string, which is also a valid optimizer value. As with all other model parameters, we leave it to the user to supply reasonable values.

### CODE REVIEW

File: usl_models/tests/model_test.py [hunk 1]
```diff
@@ -183,3 +184,38 @@ def test_bad_labels():
...
+def test_early_stopping():
+    """Tests early stopping during model training.
+
+    Since inputs are random, the model will not continually improve (there's no real
+    training). Setting a reasonably large number of epochs makes it likely that the
+    validation loss stops improving before training is complete. Enabling early stopping
+    with a patience of 1 epoch means that the model should stop training as soon as the
+    validaton loss goes up.
+
+    Note that this test is nondeterministic and possibly flaky. But with a large enough
```
Comment (waltaskew) [inline usl_models/tests/model_test.py:198]: If we set a random seed such that we're consistently using the same "random" inputs every time, I think that would address possible flakiness?
https://www.tensorflow.org/api_docs/python/tf/random/set_seed

Comment (tatianawu) [reply usl_models/tests/model_test.py]: You're right! Thanks for the suggestion, added a seed.

**Review by waltaskew — APPROVED** [review]
LGTM
